### PR TITLE
👮‍♀️ Add Typescript

### PR DESCRIPTION
**Pull request by ThibaudAV**
**Issue**

🎄 🎁 

**Description**

- Compile all the code with tsc
- No more js or html minification. it's up to the parent lib to make the choice or not

CI :
- Fix prerelease to use dist instead src
- remove unnecessary build dist already done before

**Additional context**

Test with :
- [x] Apim
- [ ] Cockpit
Maybe AM but if it's work for cockpit, no reason for AM

---

📚&nbsp;&nbsp;View the storybook of this branch [here](https://5ffff84833d7150021078521-znnqdhqwpz.chromatic.com)

### CODE REVIEW

**Comment by graviteeio** [comment]
Hey dear reviewer, I'm the Gravitee.io bot :robot:

I just want to tell you that `package.json` has been updated in this PR. 

Some components are based on third party libraries, for integration into a web component, we extract the css from its libraries as assets.

WARNING: When `highlight.js` or `github-markdown-css` are updated, we must run `npm run build` task to update the css.

<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against ad6021f4b7802d053dff51b2098f31e491680b1c
</p>

**Comment by graviteeio** [comment]
**Gravitee.io Automatic Deployment**

A beta version of this package has been published! 🚀

To use it in your project run the following lines:
```shell
npm install @gravitee/ui-components@3.25.3-typescript-ad6021f
```

**Comment by graviteeio** [comment]
**Gravitee.io Automatic Deployment**

A beta version of this package has been published! 🚀

To use it in your project run the following lines:
```shell
npm install @gravitee/ui-components@3.25.3-typescript-8a53edf
```

**Comment by graviteeio** [comment]
**Gravitee.io Automatic Deployment**

A beta version of this package has been published! 🚀

To use it in your project run the following lines:
```shell
npm install @gravitee/ui-components@3.25.3-typescript-895ea28
```

**Comment by graviteeio** [comment]
**Gravitee.io Automatic Deployment**

A beta version of this package has been published! 🚀

To use it in your project run the following lines:
```shell
npm install @gravitee/ui-components@3.25.3-typescript-9a4271f
```

**Comment by graviteeio** [comment]
**Gravitee.io Automatic Deployment**

A beta version of this package has been published! 🚀

To use it in your project run the following lines:
```shell
npm install @gravitee/ui-components@3.25.3-typescript-5a46c7d
```

**Comment by graviteeio** [comment]
**Gravitee.io Automatic Deployment**

A beta version of this package has been published! 🚀

To use it in your project run the following lines:
```shell
npm install @gravitee/ui-components@3.25.3-typescript-bfd8f67
```

**Comment by graviteeio** [comment]
**Gravitee.io Automatic Deployment**

A beta version of this package has been published! 🚀

To use it in your project run the following lines:
```shell
npm install @gravitee/ui-components@3.25.3-typescript-e7256ca
```

**Comment by graviteeio** [comment]
**Gravitee.io Automatic Deployment**

A beta version of this package has been published! 🚀

To use it in your project run the following lines:
```shell
npm install @gravitee/ui-components@3.25.3-typescript-50960c3
```

**Comment by graviteeio** [comment]
**Gravitee.io Automatic Deployment**

A beta version of this package has been published! 🚀

To use it in your project run the following lines:
```shell
npm install @gravitee/ui-components@3.25.3-typescript-fdacf14
```

**Review by gaetanmaisse — COMMENTED** [review]
👍🏻 💪🏻 

2 things we can discuss: 
 - What do you think of moving the stories next to the source files? instead of having a dedicated folder 
 - We need to check if modification (and so rollup) is really needed

**Comment by ThibaudAV** [comment]
Because I was afraid that if I revolutionize too many things at once it would not be liked. :tavenier-chasing-issues: 
Especially since "I arrive" with yarn after 😁 

but yes really agree to add the test + stories next to their sources

**Comment by graviteeio** [comment]
Hey dear reviewer, I'm the Gravitee.io bot :robot:

I just want to tell you that `package.json` has been updated in this PR. 

Some components are based on third party libraries, for integration into a web component, we extract the css from its libraries as assets.

WARNING: When `highlight.js` or `github-markdown-css` are updated, we must run `npm run build` task to update the css.

<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 94e811387f42edd37553f000935b551b3871bdd9
</p>

**Comment by graviteeio** [comment]
**Gravitee.io Automatic Deployment**

A beta version of this package has been published! 🚀

To use it in your project run the following lines:
```shell
npm install @gravitee/ui-components@3.25.3-typescript-94e8113
```

**Comment by graviteeio** [comment]
**Gravitee.io Automatic Deployment**

A beta version of this package has been published! 🚀

To use it in your project run the following lines:
```shell
npm install @gravitee/ui-components@3.25.3-typescript-9cbc31e
```

**Comment by graviteeio** [comment]
**Gravitee.io Automatic Deployment**

A beta version of this package has been published! 🚀

To use it in your project run the following lines:
```shell
npm install @gravitee/ui-components@3.25.3-typescript-9d19b68
```

**Comment by graviteeio** [comment]
**Gravitee.io Automatic Deployment**

A beta version of this package has been published! 🚀

To use it in your project run the following lines:
```shell
npm install @gravitee/ui-components@3.25.3-typescript-493e1dd
```

**Comment by graviteeio** [comment]
**Gravitee.io Automatic Deployment**

A beta version of this package has been published! 🚀

To use it in your project run the following lines:
```shell
npm install @gravitee/ui-components@3.25.3-typescript-71be1b1
```

**Comment by graviteeio** [comment]
**Gravitee.io Automatic Deployment**

A beta version of this package has been published! 🚀

To use it in your project run the following lines:
```shell
npm install @gravitee/ui-components@3.25.3-typescript-89a8132
```

File: .circleci/config.yml [hunk 1]
```diff
@@ -90,19 +90,50 @@ jobs:
...
       - checkout
       - attach_workspace:
           at: .
       - secrethub/exec:
           step-name: Running Chromatic
-          command: npx chromatic --project-token=$CHROMATIC_PROJECT_TOKEN --exit-once-uploaded -d=storybook-static
+          # TODO:
+          #  - Handle npx chromatic command failure, make the job fails
+          command: |
+            SB_URL=$(cd gravitee-apim-console-webui && npx chromatic --project-token=$CHROMATIC_PROJECT_TOKEN --exit-once-uploaded -d=storybook-static | grep -o "View your Storybook at https:\/\/[0-9a-z-]*\.chromatic\.com" | grep -o "https:.*")
```
Comment (gaetanmaisse) [inline .circleci/config.yml:106]: `cd gravitee-apim-console-webui` to remove

**Comment by graviteeio** [comment]
**Gravitee.io Automatic Deployment**

A beta version of this package has been published! 🚀

To use it in your project run the following lines:
```shell
npm install @gravitee/ui-components@3.25.3-typescript-dc20727
```

**Comment by graviteeio** [comment]
**Gravitee.io Automatic Deployment**

A beta version of this package has been published! 🚀

To use it in your project run the following lines:
```shell
npm install @gravitee/ui-components@3.25.3-typescript-4424877
```

**Comment by graviteeio** [comment]
**Gravitee.io Automatic Deployment**

A beta version of this package has been published! 🚀

To use it in your project run the following lines:
```shell
npm install @gravitee/ui-components@3.25.3-typescript-b6b9b43
```

**Comment by graviteeio** [comment]
**Gravitee.io Automatic Deployment**

A beta version of this package has been published! 🚀

To use it in your project run the following lines:
```shell
npm install @gravitee/ui-components@3.25.3-typescript-b5c0bcb
```

**Comment by graviteeio** [comment]
**Gravitee.io Automatic Deployment**

A beta version of this package has been published! 🚀

To use it in your project run the following lines:
```shell
npm install @gravitee/ui-components@3.25.3-typescript-6f4b14e
```

**Comment by graviteeio** [comment]
**Gravitee.io Automatic Deployment**

A beta version of this package has been published! 🚀

To use it in your project run the following lines:
```shell
npm install @gravitee/ui-components@3.25.3-typescript-6a73008
```

**Review by gaetanmaisse — APPROVED** [review]

**Pull request merged**